deployment/postgres-init-db/sql/views/merlin/derived_events.sql

```diff
@@ -0,0 +1,106 @@
+    output.duration,
+    output.start_time,
+    output.source_range,
+    output.valid_at
+from (
+  -- select the events from the sources and include them as they fit into the ranges determined by sub
+  select
+    s.key as source_key,
+    ee.key as event_key,
+    ee.event_type_name,
+    ee.duration,
+    s.derivation_group_name,
+    ee.start_time,
+    s.source_range,
+    s.valid_at
+  from merlin.external_event ee
+  join (
+    with base_ranges as (
+      -- base_ranges orders sources by their valid time
+      -- and extracts the multirange that they are stated to be valid over
+      select
+        external_source.key,
+        external_source.derivation_group_name,
+        tstzmultirange(tstzrange(external_source.start_time, external_source.end_time)) as range,
+        external_source.valid_at
+      from merlin.external_source
+      order by external_source.valid_at
+    ), base_and_sub_ranges as (
+      -- base_and_sub_ranges takes each of the sources above and compiles a list of all the sources that follow it
+      -- and their multiranges that they are stated to be valid over
+      select
+        base.key,
+        base.derivation_group_name,
+        base.range                                                      as original_range,
+        array_remove(array_agg(subsequent.range order by subsequent.valid_at), NULL) as subsequent_ranges,
+        base.valid_at
+      from base_ranges base
+      left join base_ranges subsequent
+        on base.derivation_group_name = subsequent.derivation_group_name
+        and base.valid_at < subsequent.valid_at
+      group by base.key, base.derivation_group_name, base.valid_at, base.range
+    )
+    -- this final selection (s) utilizes the first, as well as merlin.subtract_later_ranges,
+    -- to produce a sparse multirange that a given source is valid over.
+    -- See merlin.subtract_later_ranges for further details on subtracted ranges.
+    select
+      r.key,
+      r.derivation_group_name,
+      merlin.subtract_later_ranges(r.original_range, r.subsequent_ranges) as source_range,
+      r.valid_at
+    from base_and_sub_ranges r
+    order by r.derivation_group_name desc, r.valid_at) s
+  on s.key = ee.source_key
+  and s.derivation_group_name = ee.derivation_group_name
+  where s.source_range @> ee.start_time
+  order by valid_at desc
+) output;
+
+-- create a unique index, which allows concurrent refreshes
+create unique index on merlin.derived_events (
+  event_key,
+  source_key,
+  derivation_group_name,
+  event_type_name
+);
+
+-- refresh the materialized view after insertion/update/deletion to external_source and external_event and derivation_group
+create function merlin.refresh_derived_events_on_trigger()
+  returns trigger
+  language plpgsql as $$
+begin
+  refresh materialized view concurrently merlin.derived_events;
+  return new;
+end;
+$$;
+
+-- events are the most basic source of information, so we have to trigger on their insertion
```

Comment:
Minor, tweak comment, as it's not just an insertion the trigger (feel free to alter wording, this is just one way to tweak it)
```suggestion
-- events are the most basic source of information, so update when the set of events changes.
```